### PR TITLE
Refactoring the folders structure

### PR DESCRIPTION
**Pull request by drew2a**
This PR implements folders' restructuration that was described in #6505

To check the new folder structure, go to https://github.com/Tribler/tribler/tree/refactoring/folders_restructure

### CODE REVIEW

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

**Comment by drew2a** [comment]
retest this please

File: scripts/exit_node/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Exit Node
+
+## Requirements
+
+* Export the `src` directory to to the `PYTHONPATH`
```
Comment (kozlovsky) [inline scripts/exit_node/README.md:5]: Typo
```suggestion
* Export the `src` directory to `PYTHONPATH`
```

File: src/README.md [hunk 2]
```diff
@@ -25,17 +25,11 @@ from` requirements.txt`.
 
 ##
 
-Export to PYTHONPATH the following directories:
+Export the `src` directory to to the `PYTHONPATH`:
```
Comment (kozlovsky) [inline src/README.md:28]: Typo
```suggestion
Export the `src` directory to `PYTHONPATH`
```

**Review by kozlovsky — CHANGES_REQUESTED** [review]
Great work!

There are places where `pyipv8` references should be removed as well (remnants from #6757). I cannot specify them in the review, as these files were moved without change and the diff is unavailable, but you can find them with "pyipv8" search. I mean `src/tribler/core/__init__.py` and six examples in the `srcipts` folder. In these lines, `tribler-core` is also mentioned and should be removed, so logically this fix belongs to the current PR.

File: src/tribler/gui/upgrade_manager.py [hunk 3]
```diff
@@ -109,7 +110,6 @@ def start(self):
 
         self._upgrade_worker = StateDirUpgradeWorker(self.version_history.root_state_dir)
         # We import it here because it is safer to do it in the main thread
```
Comment (kozlovsky) [inline src/tribler/gui/upgrade_manager.py:112]: The comment should be deleted as well
```suggestion
```

**Review by kozlovsky — CHANGES_REQUESTED** [review]

**Review by kozlovsky — APPROVED** [review]

**Pull request merged**